### ISSUE TITLE
--output flag returns incorrect filename with conda 3

### ISSUE THREAD

**Issue by astrofrog** [conda/conda-build#2189]
I run conda-build on Travis to build a few packages, and I rely on

```
conda build --python <version> <package> --output
```

to determine the name of the generated file. However, it seems that in conda 3, the ``--python`` value is ignored. Here is a build with conda 2:

https://travis-ci.org/glue-viz/conda-dev/jobs/250895805

and the same build 24 hours later with conda 3:

https://travis-ci.org/glue-viz/conda-dev/jobs/251283437

The end of the conda 3 build is:

```
...
TEST END: /home/travis/miniconda/conda-bld/linux-64/glue-core-0.11.0.dev20170707163327.65d6524-py27hdd69ff2_0.tar.bz2
# Automatic uploading is disabled
# If you want to upload package(s) to anaconda.org later, type:

anaconda upload /home/travis/miniconda/conda-bld/linux-64/glue-core-0.11.0.dev20170707163327.65d6524-py27hdd69ff2_0.tar.bz2

# To have conda build upload to anaconda.org automatically, use
# $ conda config --set anaconda_upload yes

anaconda_upload is not set.  Not uploading wheels: []

travis_time:end:29e8b2d4:start=1499456108699972696,finish=1499456263982439106,duration=155282466410
[0K
[32;1mThe command "$CONDA_BUILD glue-core" exited with 0.[0m
travis_time:start:15e065a4
[0K$ $CONDA_BUILD glue-core --output
Attempting to finalize metadata for glue-core
INFO conda_build.metadata:finalize_outputs_pass(607): Attempting to finalize metadata for glue-core
/home/travis/miniconda/conda-bld/linux-64/glue-core-0.11.0.dev20170707163327.65d6524-py36h3baf84d_0.tar.bz2
...
```

As you can see, the filename returned initially by conda build disagrees with the one returned by the same command with the ``--output`` flag.

**Comment by msarahan** [comment]
Thanks for the report.  It seems that --output is ignoring your --python setting somehow.  I will investigate.

**Comment by msarahan** [comment]
This was the same bug as #2181 #2169 - this is fixed in 3.0.4, which became available yesterday evening.  Closing this issue, but if you still have problems, please let us know.
